### PR TITLE
#214 Restructure hosting of test vectors

### PR DESCRIPTION
**Pull request by diehuxx**
Separate test vectors into one test vector per file. Set up a validation script to ensure all test vectors follow a common format.

See `hosted/test-vectors/protocol/README.md` for how to use and generate test vectors

See https://github.com/TBD54566975/tbdex-js/pull/129 for how implementations can use git submodules to consume test vectors and other hosted materials.

### CODE REVIEW

**Comment by diehuxx** [comment]
Instead of using this entire repo as a submodule in implementation repos (tbdex-js, tbdex-kt, etc.), let's pull the JSON schemas and test vectors into their own repo called tbdex-interop-suite, which is included as a submodule here and in implementation repos.

**Comment by KendallWeihe** [comment]
Nice! A coupla thoughts...

- Should we also move [`scripts/test-vector-validation`](https://github.com/TBD54566975/tbdex/tree/vectors-214/scripts/test-vector-validation)?
- The [GitHub Workflow](https://github.com/TBD54566975/tbdex/blob/vectors-214/.github/workflows/publish-hosted-resources.yml) still relies on the `hosted/` directory, is this a bug?

**Comment by KendallWeihe** [comment]
> Nice! A coupla thoughts...
> 
> * Should we also move [`scripts/test-vector-validation`](https://github.com/TBD54566975/tbdex/tree/vectors-214/scripts/test-vector-validation)?
> * The [GitHub Workflow](https://github.com/TBD54566975/tbdex/blob/vectors-214/.github/workflows/publish-hosted-resources.yml) still relies on the `hosted/` directory, is this a bug?

As I was typing that out, I see you committed the files back. Sorry I may be throwing this for a loop.

File: hosted/test-vectors/protocol/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "scripts": {
+    "validate-vectors": "node validate-vectors.js"
```
Comment (nitro-neal) [inline hosted/test-vectors/protocol/package.json:3]: nice

**Review by nitro-neal — APPROVED** [review]
Awesome, once these are all merged in happy to get your help to get web5-* submodule created and working in the same fashion.

Also very soon I think the unit test xml will get upload for tbdex-* and we can add them to the checkmark page in sdk-development

**Review by mistermoe — APPROVED** [review]
💯

**Pull request merged**